cli/command/stack/kubernetes/convert.go

```diff
@@ -216,15 +243,34 @@ func fromComposeConfigs(s map[string]composeTypes.ConfigObjConfig) map[string]la
 	return m
 }
 
-func fromComposeServiceConfig(s composeTypes.ServiceConfig) latest.ServiceConfig {
-	var userID *int64
+func fromComposeServiceConfig(s composeTypes.ServiceConfig, capabilities composeCapabilities) (latest.ServiceConfig, error) {
+	var (
+		userID     *int64
+		pullSecret string
+		pullPolicy string
+		err        error
+	)
 	if s.User != "" {
 		numerical, err := strconv.Atoi(s.User)
 		if err == nil {
 			unixUserID := int64(numerical)
 			userID = &unixUserID
 		}
 	}
+	pullSecret, err = resolveServiceExtra(s, PullSecretExtraField)
+	if err != nil {
+		return latest.ServiceConfig{}, err
+	}
+	pullPolicy, err = resolveServiceExtra(s, PullPolicyExtraField)
+	if err != nil {
+		return latest.ServiceConfig{}, err
+	}
+	if pullSecret != "" && !capabilities.hasPullSecrets {
+		return latest.ServiceConfig{}, errors.Errorf("stack API version %s does not support pull secrets (field %q), please use version v1alpha3 or higher", capabilities.apiVersion, PullSecretExtraField)
```

Comment:
Wondering; why would I have to manually provide a pull-secret? Would it be possible to automatically resolve credentials for an image and pass them as a pull-secret?

Reply:
This has been discussed with @justincormack here: https://github.com/docker/compose-on-kubernetes/issues/27 and the current Swarm seem not to be a very good practice (using user credentials instead of a custom service account)